homeassistant/components/music_assistant/media_player.py

```diff
@@ -276,22 +276,36 @@ async def async_on_update(self) -> None:
             self._attr_state = MediaPlayerState(player.state.value)
         else:
             self._attr_state = MediaPlayerState(STATE_OFF)
+
         group_members_entity_ids: list[str] = []
+        coordinator_id: str | None = None
+        group_childs: set[str] = set()
         if player.group_childs:
+            # This player is the coordinator, with children players
+            coordinator_id = self.player_id
+            group_childs = set(player.group_childs) - {self.player_id}
+        elif player.synced_to:
+            # This player is a child player, with a coordinator
+            coordinator_id = player.synced_to
+            # Do not list siblings. It's unclear if we should list all siblings or just this player
+            group_childs = {self.player_id}
+
+        if coordinator_id and group_childs:
+            # Set the coordinator as the first entity by convention
+            group_members = [coordinator_id, *sorted(group_childs)]
```

Comment:
let's keep the original sort order so:

for a group leader: just use the group_childs attribute directly without modification (just translate into ha entity id's)

for a group child: lookup the group_leader (if synced_to is set) and then use that group_childs list as-is.

Reply:
All set!